code/controllers/subsystem/processing/networks.dm

```diff
@@ -1,51 +1,135 @@
+#define DEBUG_NETWORKS
+
 PROCESSING_SUBSYSTEM_DEF(networks)
 	name = "Networks"
 	priority = FIRE_PRIORITY_NETWORKS
 	wait = 1
 	stat_tag = "NET"
 	flags = SS_KEEP_TIMING
 	init_order = INIT_ORDER_NETWORKS
+
 	var/datum/ntnet/station/station_network
-	var/assignment_hardware_id = HID_RESTRICTED_END
-	var/list/networks_by_id = list()				//id = network
-	var/list/interfaces_by_id = list()				//hardware id = component interface
-	var/resolve_collisions = TRUE
+	var/list/network_initialize_queue = list()
+
+	var/list/interfaces_by_hardware_id = list()
+	var/list/networks = list()
+	// Used with map tags to look up hardware address
+
+/datum/controller/subsystem/processing/stat_entry(msg)
+	msg = "[stat_tag]:[length(processing)]"
+	return ..()
+
 
 /datum/controller/subsystem/processing/networks/Initialize()
 	station_network = new
-	station_network.register_map_supremecy()
-	. = ..()
 
-/datum/controller/subsystem/processing/networks/proc/register_network(datum/ntnet/network)
-	if(!networks_by_id[network.network_id])
-		networks_by_id[network.network_id] = network
-		return TRUE
-	return FALSE
+	return ..()
+
+/datum/controller/subsystem/processing/networks/fire(resumed = 0)
+	// so life sucks.  Can't be in Initialize because Initialize is run async and we must start
+	// when everything is built and working.
+	if(SSmapping.initialized)
+		station_network.register_map_supremecy()
+		if(network_initialize_queue.len)
+			for(var/datum/component/ntnet_interface/conn in network_initialize_queue)
+				if(conn.network) // we really should runtime if there is no network
+					SEND_SIGNAL(conn.parent, COMSIG_COMPONENT_NTNET_JOIN_NETWORK, conn.network)
+			network_initialize_queue.Cut()
+		flags |= SS_NO_FIRE // never have to run again yea!
+	else
+		to_chat(world, "Holly fuck those maps take a while to load")
+
+
+// verifies the network name has the right letters in it.  Might need to expand unwanted punctuations
+
+/datum/controller/subsystem/processing/networks/proc/verify_network_name(name)
+	return istext(name) && length(name) > 0 && name[1] != "." && name[length(name)] != "."&& findtext(name,"\[ \n-;:\]*") == 0
```

Comment:
If it's alphanumerics in some specific format only do whitelist with regex instead of this.